### ISSUE TITLE
Does `lexbor` support contains query ?

### ISSUE THREAD

**Issue by baseplate-admin** [lexbor/lexbor#181]
Cross posted from https://github.com/rushter/selectolax/issues/86#issue-1674227438

Hi Thanks for creating this awesome library.

 

I have run into an edge case where i need to check if span element contains some special type of text.

I was wondering if `lexbor` supports `span:contains('Hello')` type queries.
 Since selectolax threw `segmentation` error, i was wondering if it has to do more with `lexbor`

**Comment by lexborisov** [comment]
@baseplate-admin 

This class is not in [the specification](https://drafts.csswg.org/selectors/).
I'm not sure it should be done with CSS Selectors.

**Comment by baseplate-admin** [comment]
>  I'm not sure it should be done with CSS Selectors.

Fair enough. I will come up with some argument ( if i find any in future )..

**Comment by AmericanY** [comment]
> @baseplate-admin
> 
> This class is not in [the specification](https://drafts.csswg.org/selectors/). I'm not sure it should be done with CSS Selectors.

as of today, is there no way to find by text ? Thanks in advance  Alexander.

**Comment by lexborisov** [comment]
Hi @AmericanY 

> as of today, is there no way to find by text ?

You can't use CSS Selectors for this.

Adding something like `:contains` is not difficult. It's just not clear in what form it should be and whether it should be in CSS Selectors at all.

**Comment by spicychickensauce** [comment]
Hi there 👋 
Could this be reconsidered?

[Phoenix](https://www.phoenixframework.org/) recently switched to using [lazy_html](https://hexdocs.pm/lazy_html/LazyHTML.html) in it's testing library, which uses Lexbor.

Previously, phoenix was using [floki](https://hexdocs.pm/floki/readme.html), which supports a [`:fl-contains('foo')`](https://hexdocs.pm/floki/readme.html#supported-selectors) selector.

Where previously I could write this single selector: `li:has(div:fl-contains('foo'))`, I now need to tediously split this query into multiple parts to accomplish the same thing.

I propose to follow floki (and [jquery](https://api.jquery.com/contains-selector/) and others) here and add such a selector even if not supported by browsers:

| Selector | Description |
| --- | --- |
| :lexbor-contains('foo') |	an E element that contains "foo" inside a text node |
| :lexbor-icontains('foo') |	an E element that contains "foo" inside a text node (case insensitive) |



What do you think? Is there any chance this could be added?

**Comment by spicychickensauce** [comment]
Wow, that's awesome and so quick 😍 
Your implementation makes sense to me, given that there is no spec.

I tried testing it via `lazy_html` but couldn't get past cmake issues -.-

But I could verify it works as expected running tests from lexbor directly.
I've added this additional test case which is close to my real use case and that works as expected 👍 
```
    {"p:has(a:lexbor-contains('a3'))",
     "<p p=\"3\">"}
```

**Comment by jonatanklosko** [comment]
Hey @lexborisov! Would it be possible to release it as a new version, or are you expecting one in the near future? Technically we can pin to any revision of the code, but referencing a version seems more robust.

**Comment by lexborisov** [comment]
Hi @jonatanklosko 

I don't really want to release a new version just for one feature. I usually accumulate a certain amount of changes and then release a new version.
If it's really important, then I can release it.

**Comment by spicychickensauce** [comment]
Hey all. Sorry if I've caused a commotion. I didn't mean to cause any sense of urgency, this can wait as long as it takes.

I'm in no rush to update my code, so this can just wait until you decide to release the next minor versions, it's all good.

Thanks you both for your great work!
